routers/private/hook_post_receive.go

```diff
@@ -110,6 +110,19 @@ func HookPostReceive(ctx *gitea_context.PrivateContext) {
 			ctx.JSON(http.StatusInternalServerError, private.HookPostReceiveResult{
 				Err: fmt.Sprintf("Failed to Update: %s/%s Error: %v", ownerName, repoName, err),
 			})
+			return
+		}
+
+		// handle manually merge pulls confirm
+		mergePullIndexes := opts.GitPushOptions.Int64Array(private.GitPushOptionMergePulls)
+		if len(mergePullIndexes) > 0 {
+			if err := issues_model.ManuallyMergePullConfirmByIndexes(ctx, repo.ID, mergePullIndexes); err != nil {
+				log.Error("Failed to manually merge pull confirm: %s/%s Error: %v", ownerName, repoName, err)
+				ctx.JSON(http.StatusInternalServerError, private.HookPostReceiveResult{
+					Err: fmt.Sprintf("Failed to manually merge pull confirm: %s/%s Error: %v", ownerName, repoName, err),
+				})
+				return
+			}
```

Comment:
this doesn't appear to do any permissions checks?

Why not just do the manualMerge checks here instead of hacking in this new ManuallyMergePullConfirm flag?

Reply:
I think if do it here,  It will be duplicate with the 'pull check' logic. add a single flag to ignore `AutodetectManualMerge` config is the more simple way.

Reply:
But by not doing the check when processing the  receive-hook you risk the flag being reset to false before the pull is actually processed. Remember these two things are occurring in totally different goroutines and may be occurring semi-concurrently. 

There's no guarantee that the next time manuallyMerged() is run relates to this particular push.

Reply:
Hmm, But checking pull reuest will cost much time, should not do it in hook. it's really a hard problem ...

Reply:
@zeripath how about use version for sync? Thanks. https://github.com/go-gitea/gitea/pull/21951/commits/b8ffb3287926e3672fbae7ffcb3a3751b8dce59b